src/test/java/seedu/address/model/person/TelegramTest.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class TelegramTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Telegram(null));
+    }
+
+    @Test
+    public void constructor_invalidTelegram_throwsIllegalArgumentException() {
+        String invalidTelegram = "";
+        assertThrows(IllegalArgumentException.class, () -> new Telegram(invalidTelegram));
+    }
+
+    @Test
+    public void isValidTelegram() {
+        // null telegram handle
+        assertThrows(NullPointerException.class, () -> Telegram.isValidTelegram(null));
+
+        // invalid telegram handle
+        assertFalse(Telegram.isValidTelegram("")); // empty string
+        assertFalse(Telegram.isValidTelegram(" ")); // spaces only
+
+        // valid telegram handle
+        assertTrue(Telegram.isValidTelegram("@heyanything"));
+        assertTrue(Telegram.isValidTelegram("@s")); // one character
+        assertTrue(Telegram.isValidTelegram("@heymumble_jumbo123blahblahhellohellotestme")); // long address
```

Comment:
I believe telegram handles only accept 5-32 characters

Reply:
Resolved in 0bf8c09